sdk/ml/azure-ai-ml/azure/ai/ml/identity/_credentials/_AzureMLSparkOnBehalfOfCredential.py

```diff
@@ -0,0 +1,108 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+
+import functools
+import os
+from typing import Optional
+
+from azure.core.pipeline.transport import HttpRequest
+
+from .._internal.managed_identity_base import ManagedIdentityBase
+from .._internal.managed_identity_client import ManagedIdentityClient
+
+
+class _AzureMLSparkOnBehalfOfCredential(ManagedIdentityBase):
+    def get_client(self, **kwargs) -> Optional[ManagedIdentityClient]:
+        client_args = _get_client_args(**kwargs)
+        if client_args:
+            return ManagedIdentityClient(**client_args)
+        return None
+
+    def get_unavailable_message(self) -> str:
+        return "AzureML Spark On Behalf of credentials not available in this environment"
+
+
+def _get_client_args(**kwargs) -> Optional[dict]:
+    from pyspark.sql import SparkSession  # cspell:disable-line # pylint: disable=import-error
+
+    try:
+        spark = SparkSession.builder.getOrCreate()
+    except Exception:
+        raise Exception("Fail to get spark session, please check if spark environment is set up.")
+
+    spark_conf = spark.sparkContext.getConf()
+    spark_conf_vars = {
+        "AZUREML_SYNAPSE_CLUSTER_IDENTIFIER": "spark.synapse.clusteridentifier",
+        "AZUREML_SYNAPSE_TOKEN_SERVICE_ENDPOINT": "spark.tokenServiceEndpoint",
+    }
+    for env_key, conf_key in spark_conf_vars.items():
+        value = spark_conf.get(conf_key)
+        if value:
+            os.environ[env_key] = value
```

Comment:
The `AZUREML_SYNAPSE_CLUSTER_IDENTIFIER` and `AZUREML_SYNAPSE_TOKEN_SERVICE_ENDPOINT` just replicate the values that are already in the spark config. These values are only used by the obo token flow. We replicate these to `os.environ` so that we don't have to read from spark config each time. These are not sensitive and not PII